### ISSUE TITLE
asdf

### ISSUE THREAD

**Issue by EugenKon** [hashicorp/terraform#35252]
### Terraform Version

```shell
asdf
```

### Use Cases

asdf

### Attempted Solutions

asdf

**Comment by EugenKon** [comment]
Copied from https://github.com/hashicorp/terraform-provider-aws/issues/37701

**Comment by jbardin** [comment]
Related: #34612

**Comment by apparentlymart** [comment]
Thanks for this feature request, @EugenKon.

One thing to consider here is that by default Terraform refreshes _all_ resource instances, and so this count would really just be a total number of resource instances tracked in the prior state.

I assume what you _really_ want here is not a count of what was refreshed, but rather a count of how many instances were refreshed _and then the new state was different than the old_. For a refresh-only plan that would be the number of differences described under the "Objects have changed outside of Terraform" heading.

For a normal plan we don't show the objects that have changed outside of Terraform unless it seems likely that those changes caused at least one planned action, and so if we included a count in that case it might be confusing as to what exactly it's a count of: Terraform would not describe the items that are being counted in any other part of the output. Therefore we might choose to make the plan output also include a number of resource instances for which we detected changes outside of Terraform, even if we don't show the full details due to the filtering heuristic. However, I expect that will cause at least some of the folks that complained about earlier versions' treatment of "Objects have changed outside of Terraform" (https://github.com/hashicorp/terraform/issues/28803) to complain about Terraform showing a nonzero count of objects that changed. 

Perhaps the most appropriate compromise would be for Terraform to not mention anything about refreshing unless the plan being applied was a refresh-only plan, and then for a refresh-only plan it could mention _only_ the count of objects whose states were changed by refreshing, without including the added/changed/destroyed counts at all since there is no way for them to be nonzero in a refresh-only plan anyway. A Terraform plan "remembers" which planning mode was used to create it, so the UI layer should already have the information it needs to show a different summary depending on the planning mode.

**Comment by EugenKon** [comment]
@apparentlymart 
I agree, that you can hide added/changed/destroyed and show refreshed when `--refresh-only` option was used.

Without `--refresh-only` option you shows objects changed outside from Terraform. In that case you can in the same way summarize `refreshed` resources in the different line. Eg.:
```
Apply complete! Resources: 0 added, 0 changed, 0 destroyed.
2 refreshed.
```
Because they were shown in plan, thus should be counted in the summarize. Otherwise it is not clear what happened with those detected changes during `terraform plan` command.

**Comment by crw** [comment]
Thanks for this feature request! If you are viewing this issue and would like to indicate your interest, please use the 👍 reaction on the issue description to upvote this issue. We also welcome additional use case descriptions. Thanks again!
